python/test/assign0/conftest.py

```diff
@@ -0,0 +1,160 @@
+TEST_CODE_DIR = os.path.join(Path(__file__).parents[2], 'test/assign0')
+TIMEOUT = 30
+CHAR_WIDTH = 6
+HORIZONTAL, VERTICAL = "─", "│"
+SPACING = 10
+
+
+@unique
+class Color(Enum):
+    PURPLE = "\033[95m"
+    CYAN = "\033[96m"
+    DARKCYAN = "\033[36m"
+    BLUE = "\033[94m"
+    GREEN = "\033[92m"
+    YELLOW = "\033[93m"
+    RED = "\033[91m"
+    BOLD = "\033[1m"
+    UNDERLINE = "\033[4m"
+    END = "\033[0m"
+
+
+def compare_karel(karel1: Karel, karel2: Karel):
+    '''
+    todo: Please remove this method and the method call once Tyler Yep
+    fixes the autograder for stanfordkarel and publishes the fix to the pip
+    channel.
+    Please see https://trello.com/c/Zt4VNWa6 and https://trello.com/c/8emL7JDK
+    for corresponding workstreams
+    '''
+    if karel1 == karel2:
+        return True
+
+    """ Compares Karel Output and gets the results. """
+
+    def create_two_column_string(col1, col2):
+        """ col1 and col2 are Lists. """
+        return map(
+            lambda x: "{}{}{}".format(x[0], " " * SPACING, x[1]),
+            zip(col1, col2)
+        )
+
+    def symmetric_difference(a, b):
+        extra_a, extra_b = {}, {}
+        for k in a:
+            if k not in b:
+                extra_a[k] = a[k]
+            elif a[k] - b[k] > 0:
+                extra_a[k] = a[k] - b[k]
+        for k in b:
+            if k not in a:
+                extra_b[k] = b[k]
+            elif b[k] - a[k] > 0:
+                extra_b[k] = b[k] - a[k]
+        return extra_a, extra_b
+
+    matching = True
+    this, that = str(karel1).split("\n"), str(karel2).split("\n")
+
+    if len(this) != len(that):
+        matching = False
+    else:
+        for i in range(len(this)):
+            if this[i] != that[i]:
+                matching = False
+                break
+
+    world_width = len(this[0])
+
+    header1, header2 = "Student Output:", "Expected Output:"
+    text_spacing = " " * (world_width - len(header1) + SPACING + 1)
+    two_columns = create_two_column_string(this, that)
+    output = "\n".join(two_columns)
+    fancy_arrows = "{}❯{}❯{}❯ ".format(
+        Color.RED.value, Color.YELLOW.value, Color.GREEN.value
+    )
+
+    result = "\n\n{} {}{}{}" "\n{}{}{}\n{}\n".format(
+        fancy_arrows,
+        Color.YELLOW.value,
+        karel1.world.world_name,
+        Color.END.value,
+        header1,
+        text_spacing,
+        header2,
+        output,
+    )
+
+    if karel1.avenue != karel2.avenue or karel1.street != karel2.street:
+        matching = False
+        result += (
+            "Karel did not end up in the same location in both worlds:\n"
+            "Student: {}\n"
+            "Expected: {}\n\n".format((karel1.avenue, karel1.street),
+                                      (karel2.avenue, karel2.street))
+        )
+
+    # remove record of beepers that have a count of zero
+    for d in [karel1.world.beepers, karel2.world.beepers]:
```

Comment:
This is a bug in Tyler's `stanfordkarel` source code that I patched in lines 119-123. Please go and submit a bug fix on his code base when you have a chance.